src/azinsights/collector/MetricCollectorFactory.py

```diff
@@ -0,0 +1,29 @@
+from collector.PrometheusMetricCollector import PrometheusMetricCollector
+
+import logging
+
+logger = logging.getLogger(__name__)
+
+class MetricCollectorFactory:
+    @staticmethod
+    def Factory(collector_type: str = 'Prometheus'):
+        '''Provides instance of a certain collector type.
+        
+        Parameters
+        ----------
+        collector_type : str
+            A string representing the name of the metric collector 
+
+        Returns
+        -------
+        MetricCollector
+            An instance of a subclass of MetricCollector
+        
+        '''
+        logger.debug('Factory')
+        collectors = {
+            'Prometheus': PrometheusMetricCollector
+        }
+        if collector_type not in collectors:
+            raise Exception('{0} is not a known collector in MetricCollectorFactory.'.format(collector_type))
```

Comment:
Because this method essentially serves as a constructor for children of the MetricCollector class, I believe the correct behavior here should be to terminate execution.

Reply:
At the termination is there some sort of message or log

Reply:
If azinsights_main is running in the background, there would not be any message or log visible. However, if it were to be started alongside Moneo manually, the message would be visible. 
Logging errors can be an area of future work for this and the exporters in the worker nodes.